packages/nuxt/src/core/nuxt.ts

```diff
@@ -50,18 +51,19 @@ import { VirtualFSPlugin } from './plugins/virtual'
 
 export function createNuxt (options: NuxtOptions): Nuxt {
   const hooks = createHooks<NuxtHooks>()
```

Comment:
we probably need to ensure that hooks are called in a way that passes the name of the Nuxt concerned.... maybe via a custom caller, or by overriding `callHook`?

Reply:
hooks is assigned to the Nuxt instance.
If users wants to know which nuxt is concerned, they probably knows it when calling `useNuxt` or `nuxt.hook` <-- means they have stored it in a variable somewhere

What are you thinking of :eyes: ?

Reply:
I mean we want this to work:

```ts
nuxt.hook('modules:done', () => {
  addTemplate(someTemplate)
})
```

Reply:
How about 
```ts
  const nuxt: Nuxt = {
    _version: version,
    options,
    hooks,
    callHook: hooks.callHook,
    addHooks: hooks.addHooks,
    hook: (name, fn, opts) => asyncNameStorage.run(name, () => hooks.hook(name, fn, opts)),
    ready: () => asyncNameStorage.run(name, () => initNuxt(nuxt)),
    close: () => hooks.callHook('close', nuxt),
    vfs: {},
    apps: {},
    __name: name,
  }
```

this would only work in hooks tho.

Reply:
I think we'd still need to add a `nuxt.run` in the future

Reply:
It's also possible to acess `nuxt.hooks` directly, and in fact we use `addHooks` to add user hooks, for example.

we can have `nuxt.run`, of course, but the value of most of kit utilities is that they shouldn't _require_ it by having to pass `nuxt` around. at least reducing the surface area of possible places where users can call it would help a lot.

Reply:
> It's also possible to acess nuxt.hooks directly, and in fact we use addHooks to add user hooks, for example.

Indeed so do we need to override it directly in `hooks` ? 

> we can have nuxt.run, of course, but the value of most of kit utilities is that they shouldn't require it by having to pass nuxt around. at least reducing the surface area of possible places where users can call it would help a lot.

It's only for cases when users wants to run it outside of `loadNuxt` or `buildNuxt`, I don't really see a lot of cases 🤔